bitcoin/src/pow.rs

```diff
@@ -0,0 +1,1537 @@
+                    write!(f, "0x")?;
+                }
+                write!(f, $fmt, self.0)?;
+                write!(f, $fmt, self.1)?;
+                Ok(())
+            }
+        }
+    };
+}
+impl_hex!(LowerHex, "{:032x}");
+impl_hex!(UpperHex, "{:032X}");
+
+#[cfg(feature = "serde")]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+impl crate::serde::Serialize for U256 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: crate::serde::Serializer,
+    {
+        use crate::hashes::hex::ToHex;
+        let bytes = self.to_be_bytes();
+        if serializer.is_human_readable() {
+            // TODO: fast hex encoding.
+            serializer.serialize_str(&bytes.to_hex())
+        } else {
+            serializer.serialize_bytes(&bytes)
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+impl<'de> crate::serde::Deserialize<'de> for U256 {
+    fn deserialize<D: crate::serde::Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        use core::convert::TryInto;
+
+        use crate::hashes::hex::FromHex;
+        use crate::serde::de;
+
+        if d.is_human_readable() {
+            struct HexVisitor;
+
+            impl<'de> de::Visitor<'de> for HexVisitor {
+                type Value = U256;
+
+                fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    f.write_str("a 32 byte ASCII hex string")
+                }
+
+                fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+                where
+                    E: de::Error,
+                {
+                    if s.len() != 64 {
+                        return Err(de::Error::invalid_length(s.len(), &self));
+                    }
+
+                    let b = <[u8; 32]>::from_hex(s)
+                        .map_err(|_| de::Error::invalid_value(de::Unexpected::Str(s), &self))?;
+
+                    Ok(U256::from_be_bytes(b))
+                }
+
+                fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
+                where
+                    E: de::Error,
+                {
+                    if let Ok(hex) = core::str::from_utf8(v) {
+                        let b = <[u8; 32]>::from_hex(hex).map_err(|_| {
+                            de::Error::invalid_value(de::Unexpected::Str(hex), &self)
+                        })?;
+
+                        Ok(U256::from_be_bytes(b))
+                    } else {
+                        Err(E::invalid_value(::serde::de::Unexpected::Bytes(v), &self))
+                    }
+                }
+            }
+            d.deserialize_str(HexVisitor)
+        } else {
+            struct BytesVisitor;
+
+            impl<'de> serde::de::Visitor<'de> for BytesVisitor {
+                type Value = U256;
+
+                fn expecting(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+                    f.write_str("a bytestring")
```

Comment:
Nit: "a sequence of bytes" would sound better.

Reply:
Ok, that's three times in the last week I've balked at something, second guessed myself, and then its been picked up in review - I guess that means I'm learning :) Will fix since I have to rebase to fix conflicts anyways.